src/components/shared/Auth/Auth.jsx

```diff
@@ -1,45 +1,26 @@
 import React from 'react';
 import { Redirect } from 'react-router-dom';
 import { connect } from 'react-redux';
-import { authenticateUser, setPlayCard, setPlayWord } from '../../../store/playerActions';
+import { authenticateUser } from './AuthActions';
 
 export class Auth extends React.Component {
 
     componentDidMount() {
         this.props.authenticateUser();
-        this.instructPlayerOnRefresh();
-    }
-
-    instructPlayerOnRefresh() {
-        if (this.props.currentPlayer) {
-            if (this.props.status === "WAITING_FOR_CURRENT_PLAYER") {
-                this.props.setPlayCard(this.props.playedCard === 0 && !this.props.finishedRound && this.props.currentPlayer.username === this.props.cookie);
-            } else {
-                this.props.setPlayCard(this.props.playedCard === 0 && !this.props.finishedRound);
-            }
-            this.props.setPlayWord(this.props.currentPlayer.username === this.props.cookie && this.props.status === "WAITING_FOR_CURRENT_PLAYER" && !this.props.myWord && !this.props.finishedRound);
-        }
     }
 
     render() {
-        if (this.props.cookie) return this.props.render();
+        if (this.props.username) return this.props.render();
         return (<Redirect to='/' />);
     }
 }
 
 const mapStateToProps = (state) => ({
-    cookie: state.playerReducer.cookie,
-    currentPlayer: state.gameReducer.currentPlayer,
-    status: state.gameReducer.status,
-    finishedRound: state.gameReducer.finishedRound,
-    playedCard: state.playerReducer.playedCard,
-    myWord: state.playerReducer.myWord
+    username: state.authReducer.username
 });
 
 const mapDispatchToProps = (dispatch) => ({
-    authenticateUser: () => dispatch(authenticateUser()),
-    setPlayWord: (bool) => dispatch(setPlayWord(bool)),
-    setPlayCard: (bool) => dispatch(setPlayCard(bool))
+    authenticateUser: () => dispatch(authenticateUser())
```

Comment:
I don't think I like this component.

Reply:
What would you do instead?

Reply:
@Ro4052 what's a good practice to deal with redirecting on unsuccessful authorisation?

Reply:
Essentially, I don't think this is something React should care about, since it has nothing to do with rendering. A couple of options I can think of:

- You have this action dispatched whenever the page loads (maybe in some index file) and redirect via `history.push` if necessary
- You only use this `Auth` component for `Dashboard`, so you could just make the call there and redirect

One thing you can do if you want is have some sort of "loading" component. You could show that whenever the UI is waiting for some action to take place. E.g. you could show that while authenticating the user. That's another point though and would work in conjuction with one of the approaches above.

Note: My personal preference would probably be option 1. That is a preference though and I'd be happy with either.